### PR TITLE
Upgrade grpc to 1.44 for SPM and Cocoapods

### PR DESCRIPTION
**Pull request by wu-hui**

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -178,12 +178,12 @@ let package = Package(
...
       url: "https://github.com/firebase/abseil-cpp-SwiftPM.git",
-      "0.20200225.4" ..< "0.20200226.0"
+      "0.20220203.1" ..< "0.20220204.0"
     ),
     .package(
       name: "gRPC",
-      url: "https://github.com/firebase/grpc-SwiftPM.git",
-      "1.28.4" ..< "1.29.0"
+      url: "https://github.com/grpc/grpc-ios.git",
+      "1.44.0-grpc" ..< "1.44.1-grpc"
```
Comment (paulb777) [inline Package.swift:186]: Should be upper bound of 1.45.0-grpc to be consistent with CocoaPods to allow patch version updates for both

File: .github/workflows/firestore.yml [hunk 2]
```diff
@@ -202,7 +202,8 @@ jobs:
       run: |
         scripts/third_party/travis/retry.sh scripts/pod_lib_lint.rb ${{ matrix.podspec }} \
             --platforms=ios \
-            --allow-warnings
+            --allow-warnings \
+            --no-analyze
```
Comment (paulb777) [inline .github/workflows/firestore.yml:206]: Please open an issue to reenable analyze

Comment (wu-hui) [reply .github/workflows/firestore.yml:207]: Done.

File: Package.swift (hunk 1 above)
Comment (wu-hui) [reply Package.swift]: Done.

**Comment by dennycd** [comment]
@wu-hui  @paulb777  I realize after properly patched firestore side's usage in https://github.com/firebase/firebase-ios-sdk/pull/9539 (by matching to StrSplit's template function that does not use std::move), the error still persist, since the analysis warning also comes from building the Cocoapod abseil framework itself (see https://github.com/firebase/abseil-cpp-SwiftPM/pull/11).   https://github.com/firebase/abseil-cpp-SwiftPM/pull/11 will help resolve SPM side's building warning, but since Cocoapod is not forked from abseil-cpp-SwiftPM, we will need to upstream either a proper fix or lint-surpress  to abseil.   So I think for short term, we may need to temporarily turn off the analysis switch for this one. I will open a issue and follow up from abseil side. lmk

**Comment by wu-hui** [comment]
> @wu-hui @paulb777 I realize after properly patched firestore side's usage in #9539 (by matching to StrSplit's template function that does not use std::move), the error still persist, since the analysis warning also comes from building the Cocoapod abseil framework itself (see [firebase/abseil-cpp-SwiftPM#11](https://github.com/firebase/abseil-cpp-SwiftPM/pull/11)). [firebase/abseil-cpp-SwiftPM#11](https://github.com/firebase/abseil-cpp-SwiftPM/pull/11) will help resolve SPM side's building warning, but since Cocoapod is not forked from abseil-cpp-SwiftPM, we will need to upstream either a proper fix or lint-surpress to abseil. So I think for short term, we may need to temporarily turn off the analysis switch for this one. I will open a issue and follow up from abseil side. lmk

Thank you for your investigation! PR updated now.

File: Package.swift (hunk 1 above)
Comment (paulb777) [reply Package.swift]: I don't see it at https://github.com/firebase/firebase-ios-sdk/issues. Users should be able to find the issue if they run the analyzer on their apps that include Firestore.

File: .github/workflows/firestore.yml [hunk 3]
```diff
@@ -199,10 +199,12 @@ jobs:
       run: ./scripts/setup_bundler.sh
 
     - name: Pod lib lint
+      # TODO(b/227461966): Remove --no-analyze when absl is fixed.
```
Comment (paulb777) [inline .github/workflows/firestore.yml:202]: Should mention a public GitHub issue instead or additionally

**Review by paulb777 — APPROVED** [review]

**Pull request merged**